### PR TITLE
Bug/fix CORs

### PR DESCRIPTION
**Pull request by albert118**
## Issue

Our current CORs configuration was the "quick hack" solution to make the backend accept anything from our expected frontend server. This isn't optimal for several reasons. Namely, Vite will ~grab whatever server port is available~ spin a roulette wheel for a port, which is inconsistent for different machines.

## Fix

Since CORs is something we want universally (development, prod, etc.) it should be enabled for all environments.

The frontend should correctly proxy to the backend. In development mode this means avoiding HTTPS for now, as this is further config with a self-signed cert (which hasn't been done yet and adds more complexity).

In production, the frontend should continue to proxy to the backend but with HTTPS. Again, this requires cert configuration, which hasn't been done yet. So, this is purely to get the development environments functioning consistently.

### CODE REVIEW

File: src/frontend/vite.config.ts [hunk 1]
```diff
@@ -11,11 +11,7 @@ export default defineConfig({
     },
     server: {
         proxy: {
-            '/api': {
-                target: 'https://localhost:7230',
-                changeOrigin: false,
-                rewrite: path => path.replace(/^\/api/, '')
-            }
+            '/api': 'http://localhost:5230'
```
Comment (albert118) [inline src/frontend/vite.config.ts:14]: this shorthand is exactly what we need

File: src/server/StudioManagementSystem/Startup.cs [hunk 2]
```diff
@@ -28,16 +28,6 @@ public void ConfigureServices(IServiceCollection services, IWebHostEnvironment e
...
-
-        if (env.IsDevelopment()) {
-            services.AddCors(options => {
-                options.AddDefaultPolicy(policy => policy
-                    .WithOrigins("http://localhost:5173")
-                    .AllowAnyHeader()
-                    .AllowAnyMethod()
-                );
-            });
-        }
```
Comment (albert118) [inline src/server/StudioManagementSystem/Startup.cs:40]: we can safely remove this hack now

File: src/server/StudioManagementSystem/Startup.cs [hunk 3]
```diff
@@ -63,18 +53,17 @@ public void Configure(WebApplication app, IWebHostEnvironment env)
...
             app.UseExceptionHandler("/error")
                 // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
-                .UseHsts();
+                .UseHsts()
+                .UseHttpsRedirection();
         }
 
-        app.UseHttpsRedirection();
-
+        app.UseCors();
```
Comment (albert118) [inline src/server/StudioManagementSystem/Startup.cs:66]: CORs is configured in all environments now

File: src/server/StudioManagementSystem/Startup.cs [hunk 4]
```diff
@@ -63,18 +53,17 @@ public void Configure(WebApplication app, IWebHostEnvironment env)
...
-                .UseCors();
+                .UseDeveloperExceptionPage();
         }
         else {
             // Enable the exception handler route
             app.UseExceptionHandler("/error")
                 // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
-                .UseHsts();
+                .UseHsts()
+                .UseHttpsRedirection();
```
Comment (albert118) [inline src/server/StudioManagementSystem/Startup.cs:63]: HTTPS (redirection) is only configured outside of development environments

File: src/frontend/vite.config.ts (hunk 1 above)
Comment (albert118) [reply src/frontend/vite.config.ts:14]: For now, we also redirect to the HTTP (**not HTTPS**) backend API. Later, we can update the config for HTTPS in prod environments

**Comment by albert118** [comment]
closes #90

**Review by JaydenL33 — APPROVED** [review]

**Pull request merged**